frontend/src/metabase/containers/NewItemMenu/NewItemMenu.unit.spec.tsx

```diff
@@ -0,0 +1,99 @@
+import React from "react";
+import nock from "nock";
+import userEvent from "@testing-library/user-event";
+
+import { renderWithProviders, screen } from "__support__/ui";
+import { setupDatabasesEndpoints } from "__support__/server-mocks";
+
+import type { Database } from "metabase-types/api";
+import { createMockCard, createMockDatabase } from "metabase-types/api/mocks";
+
+import NewItemMenu from "./NewItemMenu";
+
+// eslint-disable-next-line react/display-name
+jest.mock("metabase/actions/containers/ActionCreator", () => () => (
+  <div data-testid="mock-action-editor" />
+));
+
+console.warn = jest.fn();
+console.error = jest.fn();
+
+type SetupOpts = {
+  databases?: Database[];
+  hasModels?: boolean;
+};
+
+const SAMPLE_DATABASE = createMockDatabase({
+  id: 1,
+  engine: "postgres",
+  name: "Sample Database",
+  native_permissions: "write",
+  is_sample: true,
+  settings: null,
+});
+
+const DB_WITH_ACTIONS = createMockDatabase({
+  id: 2,
+  name: "Postgres with actions",
+  engine: "postgres",
+  native_permissions: "write",
+  settings: { "database-enable-actions": true },
+});
+
+const DB_WITHOUT_WRITE_ACCESS = createMockDatabase({
+  ...DB_WITH_ACTIONS,
+  id: 3,
+  native_permissions: "none",
+});
+
+function setup({
+  databases = [SAMPLE_DATABASE, DB_WITH_ACTIONS],
+  hasModels = true,
+}: SetupOpts = {}) {
+  const scope = nock(location.origin);
+  const models = hasModels ? [createMockCard({ dataset: true })] : [];
+
+  setupDatabasesEndpoints(scope, databases);
+
+  scope.get(/\/api\/search+/).reply(200, {
```

Comment:
If I'm not mistaken this matches strings like `/api/searchhhhhhhhhh` since `+` is after `h`. I believed you want something like
```suggestion
  scope.get(/\/api\/search?.+/).reply(200, {
```

instead. Since the search endpoint should always have query parameters anyway (right?)

Or even just

```suggestion
  scope.get(/\/api\/search/).reply(200, {
```

would be fine I imo.